### PR TITLE
chore(gha): cancel in progress on prs to not clog up CI

### PR DESCRIPTION
**Pull request by slonka**
## Motivation

We're sometimes seeing long queue times and that can be partially due to people pushing to PRs often.

## Implementation information

I'm adding a concurrency group to cancel jobs in progress on PRs only.

## Supporting documentation

xrel https://github.com/kumahq/kuma/pull/10146/files

### CODE REVIEW

File: .github/workflows/build-test-distribute.yaml [hunk 1]
```diff
@@ -13,6 +13,9 @@ permissions:
 env:
   KUMA_DIR: "."
   CI_TOOLS_DIR: "/home/runner/work/kuma/kuma/.ci_tools"
+concurrency:
+  group: ${{github.workflow}}-${{ github.ref == 'refs/heads/master' && github.sha || github.event.pull_request.number }}
```
Comment (michaelbeaumont) [inline .github/workflows/build-test-distribute.yaml:17]: any reason not to just use `github.event_name == 'pull_request'`?

Comment (michaelbeaumont) [reply .github/workflows/build-test-distribute.yaml:17]: i just mean why use `github.ref == 'refs/heads/master'`? don't we really mean `event_name == 'push'`? I find that makes it easier to understand what's happening

Comment (slonka) [reply .github/workflows/build-test-distribute.yaml]: you're right

**Review by michaelbeaumont — APPROVED** [review]
LGTM let's try it

**Pull request merged**